LSPS1/LSPS1.yaml

```diff
@@ -0,0 +1,489 @@
+openapi: 3.0.3
+info:
+  description: |
+    API documentation for channel opening request from a Lightning Network Service Provider (LSP). Base API.
+    
+
+    ### Extensions
+    The api is split in to the base api and extensions.
+    All supported extensions are listed in the general API info endpoint.
+    Extensions can add additional endpoints or fields.
+
+  version: "2"
+  title: "LSP Spec 1"
+tags:
+- name: "API Info"
+- name: "Channel Request"
+
+
+paths:
+  /lsp/channels:
+    get:
+      tags:
+      - "API Info"
+      summary: "Get general API information."
+      description: |
+        Get general information about this API and supported extensions.
+        
+      operationId: "channel-order-info"
+      responses:
+        "200":
+          description: |
+            ## Extensions
+            
+            ### user_balance
+            Indicates if the LSP is willing to push satoshi to the receiver side.
+            
+            ### onchain_payments
+            Indicates if the LSP is willing to receive payments onchain.
+            
+          content:
+              application/json:
+                schema:
+                    $ref: "#/components/schemas/ApiInfoResponse"
+                example:
+                  versions: [2]
+                  website: "http://example.com/contact"
+                  options:
+                    minimum_depth: 0
+                    supports_zero_channel_reserve: true
+                    max_user_balance_satoshi: "0"
+                    max_lsp_balance_satoshi: "100000000"
+                    min_required_onchain_satoshi: null
+                    max_channel_expiry_blocks: 144
+  
+    post:
+      tags:
+      - "Channel Request"
```

Comment:
If we want to support HTTP, gRPC and onion messages, it's better to have different paths for the information and request a channel. What about  GET /lsp/channelinformation and POST /lsp/channel. I am open to any other suggestion

Reply:
I am not 100% sure yet if we should implement the API in several formats as this doubles the implementation work for LSPs. One single format would make sure everything is interoperable.

Nonetheless my concerns aside, if we go ahead with this I propose the following format:

1. General formats and requirements.
2. HTTP requirements.
3. gRPC requirements.

HTTP is supported by the OpenApi doc and gRPC should be supported by a Interface Definition Language (IDL) doc.

Let me make an example for the `GET /lsp/channels` endpoint:

## api_information_endpoint

The api_information_endpoint is the entrypoint for each client using the api. It lists the versions of the api and all options in a dictionary.

The user MUST call api_information_endpoint to read

- the `versions` of the api and therefore prove compatibility.
- `options` properties to determine the api boundaries.

Example api_information_endpoint response: 

```JSON
{
  "versions": [2],
  "website": "http://example.com/contact",
  "options": {
      "minimum_depth": 0,
      "supports_zero_channel_reserve": true,
      "max_user_balance_satoshi": "0",
      "max_lsp_balance_satoshi": "100000000",
      "min_required_onchain_satoshi": null,
      "max_channel_expiry_blocks": 20160
  }
}
```

> **HTTP** Method: GET; Path: /lsp/channels; Code 200
> Any other http details for this api_information_endpoint go here.

> **gRPC** Service: lsps1; Rpc: apiInfo
> Any other gRPC details for this api_information_endpoint go here.

Reply:
Gonna close this due to LSPS0.